# 1. ROS节点参数 =============================================================
odom_topic: "/sim/odom"
depth_topic: "/depth_image"
lidar_topic: "/lidar_points"
collision_topic: "/collision"

render_depth: true
depth_fps: 100

render_lidar: true
lidar_fps: 10

# 2. 仿真环境参数 =============================================================
random_map: true       # 使用预先建立的点云地图还是随机地图
resolution: 0.1        # 地图分辨率

# 2.1 预先建立的点云地图 (此分支已弃用)
ply_file: "src/pointcloud/forest.ply"
expand_x_times: 0      # x方向复制次数（cuda版不使用）
expand_y_times: 0      # y方向复制次数（cuda版不使用）
occupy_threshold: 0    # 大于多少个点才认为占据（cpu版不使用）

# 2.2 随机地图 ======
seed: 3
x_length: 120           # 真正随机范围(XY更大范围处为其镜像)
y_length: 30
z_length: 15
maze_type: 5           # 1: 溶洞 2: 柱子 3:迷宫 5:森林 6:房间 7:墙面  (当前策略支持1,2,5,7)
# 溶洞
complexity: 0.02       # 复杂度 0.0-0.5
fill: 0.1              # 填充率 0.0-0.4
fractal: 1
attenuation: 0.1
# 柱子
width_min: 0.6
width_max: 1.5
obstacle_number: 100
# 迷宫
road_width: 3
add_wall_x: 1
add_wall_y: 1
# 森林
tree_file: "/home/uav/YOPO_simulator_ws/Simulator/src/pointcloud/tree.ply"
tree_dist: 4
# 房间
room_number: 4         # 每个方向上的房间数 (房间大小 = x_length / room_number)
max_windows: 2         # 每面墙最多窗口数
window_size_min: 2.0
window_size_max: 2.8
add_ceiling: 0         # 是否添加天花板
# 墙面
wall_width_min: 0.5
wall_width_max: 8.0
wall_thick: 0.5
wall_number: 100       # 墙面数量
wall_ceiling: 1        # 是否添加天花板

# 3. 传感器参数 ==============================================================
camera:
  fx: 352.0             # focal length x
  fy: 352.0             # focal length y
  cx: 320.0             # principal point x (image center)
  cy: 240.0              # principal point y (image center)
  image_width: 640
  image_height: 480
  max_depth_dist: 10.0
  normalize_depth: false
  pitch: -0.0           # 相机相对机体的俯仰角 (仰为负, cpu版尚未添加)

lidar:
  vertical_lines: 16             # 纵向16线
  vertical_angle_start: -15.0    # 起始垂直角度
  vertical_angle_end: 15.0       # 结束垂直角度
  horizontal_num: 360            # 水平360点
  horizontal_resolution: 1.0     # 水平分辨率为1度，horizontal_resolution * horizontal_num = 360
  max_lidar_dist: 20.0

# 4. 数据收集参数 =============================================================
save_path: "../dataset/"
env_num: 10             # 地图数量
image_num: 10000        # 每个地图收集图像数量
roll_range: 30          # 采样点的roll角度范围
pitch_range: 30         # 采样点的pitch角度范围
x_range: 40             # x方向采样范围 (请小于地图大小x_length并保留一定边缘间距)
y_range: 40             # y方向采样范围 (请小于地图大小y_length并保留一定边缘间距)
z_range: [0.5, 4]       # z方向采样范围 (请小于地图大小z_length并保留一定边缘间距)
safe_dist: 0.5          # 安全距离，采样点与障碍的最小距离
ply_res: 0.1            # 保存点云滤波的体素网格大小
